### DISCUSSION TITLE
Allow defining allowed environment variables through the CLI

### DISCUSSION THREAD

**Discussion by KennethHoff** [vercel/turborepo#9024] [Ideas]
### Goals

1. Make it easier to migrate away from `--env-mode=loose`
2. Make it easier to integrate with orchestrators/CI platforms (In particular .Net Aspire, which relies heavily on environment variables for its integration, and was the reason I created this issue)

### Background

This is kind of the inverse of https://github.com/vercel/turborepo/pull/8757, which recently shipped.

### Proposal

Calling this command
```bash
turbo run dev --env=API1_BASE_URL --env=API2_BASE_URL
```

with this configuration:

```json
{
  "$schema": "https://turbo.build/schema.json",
  "tasks": {
    "dev": {
    }
  }
}
```

would be equivalent to this:

```json
{
  "$schema": "https://turbo.build/schema.json",
  "tasks": {
    "dev": {
      "env": ["API1_BASE_URL", "API2_BASE_URL"]
    }
  }
}
```

just like how this command
```bash
turbo run dev --env-mode=loose
```

with this configuration:

```json
{
  "$schema": "https://turbo.build/schema.json",
  "tasks": {
    "dev": {
    }
  }
}
```

is equivalent to this:

```json
{
  "$schema": "https://turbo.build/schema.json",
  "tasks": {
    "dev": {
      "envMode": "loose"
    }
  }
}
```

**Comment by anthonyshew** [comment]
Can you explain how this is different than setting up the `env` key instead?

**Reply by KennethHoff** [reply]
I assume you mean in the `turbo.json` file? Because I don't have write access to the files in the frontend repository.
